From 9d347e4d6bd40ceddf2559bafe91fe2a4d7cb6a9 Mon Sep 17 00:00:00 2001
From: tedyu <yuzhihong@gmail.com>
Date: Thu, 23 Jun 2016 07:29:34 -0700
Subject: [PATCH 405/410] HBASE-16090 ResultScanner is not closed in
 SyncTable#finishRemainingHashRanges()

(cherry picked from commit cb02b7264a4445072339149e5faf0f631f8ac20b)

Change-Id: Ida0cb8392027747389153772d7623c4c6302fc7d
Reason: Bug
Author: Ted Yu
Ref: CDH-56681
---
 .../apache/hadoop/hbase/mapreduce/SyncTable.java   |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/SyncTable.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/SyncTable.java
index 9fe4e30..8727c71 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/SyncTable.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/SyncTable.java
@@ -660,9 +660,16 @@ public class SyncTable extends Configured implements Tool {
             scan.setStopRow(nextSourceKey.copyBytes());
           }
           
-          ResultScanner targetScanner = targetTable.getScanner(scan);
-          for (Result row : targetScanner) {
-            targetHasher.hashResult(row);          
+          ResultScanner targetScanner = null;
+          try {
+            targetScanner = targetTable.getScanner(scan);
+            for (Result row : targetScanner) {
+              targetHasher.hashResult(row);
+            }
+          } finally {
+            if (targetScanner != null) {
+              targetScanner.close();
+            }
           }
         } // else current batch ends exactly at split end row
 
-- 
1.7.9.5

